app/src/main/java/com/mapbox/maps/testapp/examples/style/TileJsonActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.maps.testapp.examples.style
+
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mapbox.common.ValueConverter
+import com.mapbox.maps.MapView
+import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.extension.style.layers.addLayer
+import com.mapbox.maps.extension.style.layers.generated.rasterLayer
+import com.mapbox.maps.extension.style.sources.TileSet
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.Scheme
+import com.mapbox.maps.extension.style.sources.generated.rasterSource
+import com.mapbox.maps.testapp.R
+
+/**
+ * Activity showcases usage of TileSet to load a TileJSON compatible configuration as a source.
+ *
+ * This example uses OSM tiles as a raster source and visualises them using a raster layer.
+ */
+class TileJsonActivity : AppCompatActivity() {
+
+  private lateinit var mapboxMap: MapboxMap
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_custom_layer)
+    val mapView: MapView = findViewById(R.id.mapView)
+
+    val tileSet = TileSet.Builder("2.0.0", listOf("http://tile.openstreetmap.org/{z}/{x}/{y}.png"))
+      .name("OpenStreetMap")
+      .description("A free editable map of the whole world.")
+      .version("1.0.0")
+      .attribution("&copy; OpenStreetMap contributors, CC-BY-SA")
+      .scheme(Scheme.XYZ)
+      .minZoom(0)
+      .maxZoom(18)
+      .bounds(listOf(-180.0, -85.0, 180.0, 85.0))
+      .center(listOf(11.9, 57.7, 8.0))
+      .build()
+
+    mapboxMap = mapView.getMapboxMap()
+    mapboxMap.loadStyleJson("{}") {
+      it.addSource(
+        rasterSource(SOURCE_ID) {
+          tileSet(tileSet)
+          tileSize(256)
```

Comment:
does it mean to be strictly 256? if no perhaps move to constants so that user could play around changing that value